### DISCUSSION TITLE
Bring-up issues with jetson-xavier-nx-devkit-tx2-nx

### DISCUSSION THREAD

**Discussion by dat-tran-matternet** [OE4T/meta-tegra#625] [Q&A] [answered]
I was able to bitbake a minimal image for the new machine jetson-xavier-nx-devkit-tx2-nx  (TX2 NX). However after flashing with tegra doflash.sh the system is not able to boot into Linux. Below is what U-Boot outputs. I am running master on all meta layers. Any help would be appreciated. thanks

U-Boot 2021.01 (Feb 28 2021 - 13:04:45 +0000)

SoC: tegra186
Model: NVIDIA P3636-0001
Board: NVIDIA P3636-0001
DRAM:  3.8 GiB
MMC:   sdhci@3400000: 1, sdhci@3460000: 0
Loading Environment from MMC... *** Warning - bad CRC, using default environment

In:    serial
Out:   serial
Err:   serial
Net:   eth0: ethernet@2490000
Hit any key to stop autoboot:  0 
Card did not respond to voltage select! : -110
switch to partitions #0, OK
mmc0(part 0) is current device
Scanning mmc 0:1...
libfdt fdt_check_header(): FDT_ERR_BADMAGIC
Card did not respond to voltage select! : -110
Scanning disk sdhci@3400000.blk...
Disk sdhci@3400000.blk not ready
Scanning disk sdhci@3460000.blk...
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **
** Unrecognized filesystem type **

**Comment by koansoftware** [comment]
`Card did not respond to voltage select! : -110`
Hardware issue

**Reply by dat-tran-matternet** [reply]
Is this a sd card or emmc? I'm using the emmc for this machine. It actually doesn't have a sd card slot. Maybe it's so new they didn't populate it? Either way the emmc should be populated I thought

**Reply by ichergui** [reply]
Hey @dat-tran-matternet 

I didn't try it, I don't have yet the module `Jetson TX2 NX` but I think @madisongh does, I not sure.

**Comment by dat-tran-matternet** [comment, marked as answer]
It is working now when I use a core-image-base. I'll have to compare what I am doing wrong with my custom image. Thanks guys.

**Reply by madisongh** [reply]
Might be a device tree issue, based on the
```
libfdt fdt_check_header(): FDT_ERR_BADMAGIC
```

**Reply by dat-tran-matternet** [reply]
It's actually a really stupid mistake on my end. I had IMAGE_INSTALL instead of IMAGE_INSTALL_append. thanks everyone for your help.

**Comment by dat-tran-matternet** [comment]
I'm adding the can interface to the device tree; copying a device tree from TX2. My question is do I need the gpio_can_stb and gpio_can_en defines for the TX2 NX? Thanks

```
diff --git a/nvidia/platform/t18x/lanai/kernel-dts/tegra186-p3636-0001-p3509-0000-a01.dts b/nvidia/platform/t18x/lanai/kernel-dts/tegra186-p3636-0001-p3509-0000-a01.dts
index 06efa1ac23dc..a96c11315511 100644
--- a/nvidia/platform/t18x/lanai/kernel-dts/tegra186-p3636-0001-p3509-0000-a01.dts
+++ b/nvidia/platform/t18x/lanai/kernel-dts/tegra186-p3636-0001-p3509-0000-a01.dts
@@ -204,6 +204,12 @@
        i2c@c250000 {
                status = "okay";
        };
+
+    mttcan@c310000 {
+        status = "okay";
+        //gpio_can_stb = <&tegra_aon_gpio TEGRA_AON_GPIO(AA, 0) GPIO_ACTIVE_HIGH>;
+        //gpio_can_en = <&tegra_aon_gpio TEGRA_AON_GPIO(AA, 1) GPIO_ACTIVE_HIGH>;
+    };
 };
 #if LINUX_VERSION >= 414
 #include <tegra186-linux-4.14.dtsi>

```

**Reply by madisongh** [reply]
Based on what I see in the [pinmux configuration spreadsheet for the TX2 NX](https://developer.nvidia.com/jetson-tx2-nx-pinmux-table), it doesn't look like any of the CAN GPIOs are wired up; only CAN0 TX and RX.
